src/scripts/main.js

```diff
@@ -1,3 +1,224 @@
 'use strict';
 
-// write code here
+const DESC = 'desc';
+const ASC = 'asc';
+const HEADERS = ['name', 'position', 'office', 'age', 'salary'];
+const OFFICES = [
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+];
+
+const itemCollection = document.querySelectorAll('tbody tr');
+const tableHeader = document.querySelector('thead tr');
+const tableBody = document.querySelector('tbody');
+
+const items = [...itemCollection].map((item) => {
+  const collection = item.querySelectorAll('td');
+
+  return HEADERS.reduce(
+    (prev, current, index) => ({
+      ...prev,
+      [current]: collection[index].innerText,
+    }),
+    {},
+  );
+});
+
+function rebuildList(data) {
+  tableBody.innerHTML = '';
+
+  for (const item of data) {
+    const row = document.createElement('tr');
+
+    for (const key in item) {
+      const rowData = document.createElement('td');
+
+      rowData.innerText = item[key];
+      row.appendChild(rowData);
+    }
+
+    tableBody.appendChild(row);
+  }
+}
```

Comment:
The creation of table rows and cells can be moved to separate functions. This will make the code easier to understand and maintain.